manifests/init.pp

```diff
@@ -91,33 +91,36 @@
 ) {
   if $manage_install {
     contain letsencrypt::install # lint:ignore:relative_classname_inclusion
-    Class['letsencrypt::install'] ~> Exec['initialize letsencrypt']
     Class['letsencrypt::install'] -> Class['letsencrypt::renew']
-  }
 
-  $command = $install_method ? {
-    'package' => $package_command,
-    'vcs'     => "${venv_path}/bin/letsencrypt",
+    if $install_method == 'vcs' {
+      Class['letsencrypt::install'] ~> Exec['initialize letsencrypt']
+    }
   }
 
-  $command_init = $install_method ? {
+  $command = $install_method ? {
     'package' => $package_command,
-    'vcs'     => "${path}/letsencrypt-auto",
+    'vcs'     => "${path}/venv3/bin/certbot",
   }
 
   if $manage_config {
     contain letsencrypt::config # lint:ignore:relative_classname_inclusion
-    Class['letsencrypt::config'] -> Exec['initialize letsencrypt']
+
+    if $install_method == 'vcs' {
+      Class['letsencrypt::config'] -> Exec['initialize letsencrypt']
+    }
   }
 
   contain letsencrypt::renew
 
-  # TODO: do we need this command when installing from package?
-  exec { 'initialize letsencrypt':
-    command     => "${command_init} -h",
-    path        => $facts['path'],
-    environment => concat(["VENV_PATH=${venv_path}"], $environment),
-    refreshonly => true,
+  if $install_method == 'vcs' {
+    exec { 'initialize letsencrypt':
+      command     => 'python3 tools/venv3.py',
```

Comment:
What does this do? Isn't this sort of an install command?

Reply:
does that setup the virtualenv? Doesn't it make sense to use the voxpupuli/python module to create it?